web_registry/src/components/PatientDetail/ValuesInventory.tsx

```diff
@@ -4,51 +4,75 @@ import { Grid, Table, TableBody, TableCell, TableContainer, TableHead, TableRow,
 import { format } from 'date-fns';
 import { observer } from 'mobx-react-lite';
 import React, { FunctionComponent } from 'react';
-import { KeyedMap } from 'shared/types';
+import { IActivity } from 'shared/types';
+
 import ActionPanel, { IActionButton } from 'src/components/common/ActionPanel';
 import { getString } from 'src/services/strings';
 import { usePatient, useStores } from 'src/stores/stores';
 
 export const ValuesInventory: FunctionComponent = observer(() => {
-    const {
-        appContentConfig: { lifeAreas },
-    } = useStores();
+    const rootStore = useStores();
     const currentPatient = usePatient();
-    const { valuesInventory } = currentPatient;
-    const { assigned, assignedDateTime, lastUpdatedDateTime, values } = valuesInventory;
-
-    const lifeareaMap = Object.assign({}, ...lifeAreas.map((la) => ({ [la.id]: la.name }))) as KeyedMap<string>;
+    const { values, valuesInventory } = currentPatient;
+    const { assigned, assignedDateTime } = valuesInventory;
 
-    const activities = values
+    const activitiesWithValue = values
         ?.map((v) => {
-            return v.activities.map((a) => {
-                return {
-                    value: v.name,
-                    lifearea: lifeareaMap[v.lifeAreaId],
-                    name: a.name,
-                    enjoyment: a.enjoyment,
-                    importance: a.importance,
-                    lastEdited: a.editedDateTime,
-                };
-            });
+            return currentPatient.getActivitiesByValueId(v.valueId as string);
         })
         .reduce((a, b) => a.concat(b), []);
 
+    const otherActivites = currentPatient.getActivitiesWithoutValueId();
+
     var dateStrings: string[] = [];
     if (assigned && !!assignedDateTime) {
         dateStrings.push(
             `${getString('patient_values_inventory_assigned_date')} ${format(assignedDateTime, 'MM/dd/yyyy')}`,
         );
     }
 
-    if (!!lastUpdatedDateTime) {
-        dateStrings.push(
-            `${getString('patient_values_inventory_activity_date_header')} ${format(
-                lastUpdatedDateTime,
-                'MM/dd/yyyy',
-            )}`,
+    // NOTE: Why did we remove lastUpdatedDateTime from values-inventory.json?
```

Comment:
The field got removed from the `values inventory` document because keeping it there would have required maintaining it from anywhere that was modifying `value` or `activity` documents. We can instead calculate it.

Based on some email I traded w/ Jesse et al in September, we'll calculate this as:

- Obtain all `activity` documents
- Filter those which are not associated with any `value`
- The most recent time

I think we should then also change the value of `patient_values_inventory_activity_date_header` to be `Last updated on`.

Reply:
For to say, this seems like something we could calculate in the store.